### PR TITLE
Cleanup and add some missing calls to forceMove

### PR DESCRIPTION
**Pull request by Cyberboss**

### CODE REVIEW

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -139,21 +139,29 @@
...
+
 		loc = destination
+
 		if(!same_loc)
 			destination.Entered(src, oldloc)
-		var/area/old_area = get_area(oldloc)
-		var/area/destarea = get_area(destination)
-		if(old_area != destarea)
-			destarea.Entered(src)
+			if(destarea && oldarea != destarea)
```
Comment (XDTM) [inline code/game/atoms_movable.dm]: old_area

Comment (Cyberboss) [inline code/game/atoms_movable.dm]: instead of oldloc?

File: code/game/atoms_movable.dm [hunk 2]
```diff
@@ -139,21 +139,29 @@
 	if(destination)
 		if(pulledby)
 			pulledby.stop_pulling()
+
 		var/atom/oldloc = loc
-		var/same_loc = oldloc == destination.loc
+		var/same_loc = oldloc == destination
```
Comment (AnturK) [inline code/game/atoms_movable.dm]: Dropped the loc here ?

Comment (Cyberboss) [inline code/game/atoms_movable.dm]: Never should have been there https://github.com/tgstation/tgstation/pull/22490/files#diff-85b909bea63591ebec264bda09fbb017R145

**Pull request merged**